spine-unity/Assets/spine-unity/SkeletonAnimator.cs

```diff
@@ -202,4 +204,14 @@ public enum MixMode { AlwaysMix, MixNext, SpineStyle }
 			_UpdateComplete(this);
 		}
 	}
+
+	private int GetAnimationClipNameHashCode(AnimationClip clip) {
+		int clipNameHashCode;
+		if (!clipNameHashCodeTable.TryGetValue(clip, out clipNameHashCode)) {
```

Comment:
.NET sure does not cache the string hash code. Just looked at Mono source to be extra sure, and no, there is no caching. 
The whole hashmap thing is to avoid allocating new strings when accessing `AnimationClip.name`, so it _might_ be a tiny bit slower, but reduces the amount of generated garbage.